### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by Estherstarr**
In this PR, I implemented the following:-

- [ ] Splash screen

 - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- [ ] Sign up and log in pages

 - The user should be able to register in the app with full name, email and password (all mandatory).
 - The user can log into the app using email and password.
 - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
 - Home page (categories page)

- [ ] When the user logs in, they are presented with the categories page.
 - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
 - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
 - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
 
- [ ] Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page

- [ ] The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- [ ] Testing requirements
 - Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

**Review by MeqdamAlqudah — COMMENTED** [review]
Hi @Estherstarr  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have Linters setup correctly.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Onomeh, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has incorrect gitflow/ GitHub flow. Kindly note that capstone project worked on by a single contributor should have merge request flow **from feature into development**. Kindly use the correct gitflow and request another code review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Estherstarr** [comment]
@rloterh 
Please I'm a bit confused by your response in my project review (edited) 
I created individual branches and merged to dev


Screenshot 2023-02-25 at 12 21 13 PM


This is a screenshot of my pull requests from from feature branches to dev

**Comment by rloterh** [comment]
Hi @Estherstarr 

From the below screenshot below, you should rather have something like merge 33 commits into `dev from feature`, instead of merge 33 commits into `master from dev`

Cheers! 😃 

image

File: README.md [hunk 1]
```diff
@@ -47,187 +11,122 @@ After you're finished please remove all the comments and instructions!
...
-- **[key_feature_1]**
-- **[key_feature_2]**
-- **[key_feature_3]**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- LIVE DEMO -->
 
-## 🚀 Live Demo <a name="live-demo"></a>
+## 🚀 Live Demo <a name="https://www.loom.com/share/f851077fcf24473f98f37a8bbeb42a1f"></a>
```
Comment (KayLemba) [inline README.md]: Please kindly add deploy link to your project. Please follow the following guidelines
- The student deploys the application using Heroku, Netlify or GitHub Pages.
- The project is deployed properly, the home and detail page works as expected.
- A link to the deployed project is present in the README.

File: README.md [hunk 2]
```diff
@@ -237,52 +136,32 @@ Example:
...
 
 <!-- ACKNOWLEDGEMENTS -->
 
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
-> Give credit to everyone who inspired your codebase.
-
-I would like to thank...
+Ruby documentation 
+Railsguide documentation
```
Comment (KayLemba) [inline README.md:159]: Please kindly make sure your README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

**Review by KayLemba — CHANGES_REQUESTED** [review]
Hi @Estherstarr   👋
Good job so far working on the project requirements!
There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight 🎉
No linter errors. ✔️
GitHub flow is followed. ✔️

Required Changes ♻️
Check the comments under the review in the inline comments.
 
Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 3]
```diff
@@ -237,52 +138,32 @@ Example:
...
+Give this project a ⭐️ if you like it !
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- ACKNOWLEDGEMENTS -->
 
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
-> Give credit to everyone who inspired your codebase.
-
```
Comment (mwanawabangona) [inline README.md:259]: - [ ] `required`
- [ ] Good job on the implementing some of the requested changes
- [ ] As the previous reviewer suggested
- [ ] Please kindly make sure your README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). 🥂

File: README.md [hunk 4]
```diff
@@ -47,187 +11,124 @@ After you're finished please remove all the comments and instructions!
...
+👤 **onwuanibe Onomeh** 
 
-👤 **Author2**
-
-- GitHub: [@githubhandle](https://github.com/githubhandle)
-- Twitter: [@twitterhandle](https://twitter.com/twitterhandle)
-- LinkedIn: [LinkedIn](https://linkedin.com/in/linkedinhandle)
+- GitHub: [@githubhandle](https://github.com/Estherstarr/)
+- Twitter: [@twitterhandle](https://twitter.com/AnibeEsther)
+- LinkedIn: [LinkedIn](https://linkedin.com/in/onwuanibeonome)
```
Comment (mwanawabangona) [inline README.md]: - [ ] `Optional`
- [ ] Kindly add your `social handles`.
- [ ] Its important for employers and other developers when they want to find you on social accounts. 🥂

**Review by mwanawabangona — CHANGES_REQUESTED** [review]
Hi @Estherstarr 

Great work on making the changes requested by a previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* ✅  No linter errors :heavy_check_mark:
* ✅  Repo looks professional :briefcase:
* ✅  Good quality commits


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @Estherstarr 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  

H
## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**